app/src/hid.c

```diff
@@ -246,10 +328,17 @@ bool zmk_hid_is_pressed(uint32_t usage) {
     return false;
 }
 
-struct zmk_hid_keyboard_report *zmk_hid_get_keyboard_report() {
-    return &keyboard_report;
+uint8_t *zmk_hid_get_keyboard_report(zmk_hid_report_request_t req_t, uint8_t proto) {
+    if (proto == HID_PROTOCOL_REPORT) {
+        return req_t == HID_REPORT_FULL ? (uint8_t*)&keyboard_report : (uint8_t*)&keyboard_report.body;
```

Comment:
I feel like it would be simpler to just have separate functions for getting the full report and just the body. Every location this gets used, the value of `req_t` is hard-coded. Similarly, there's only one place that actually dynamically sets the value of `proto`. 

Maybe it would be better to keep these functions returning typed pointers, and in the one place that needs to switch on the protocol, just do something like this?

```c
void* data;
size_t size;
if (protocol == HID_PROTOCOL_REPORT) {
  data = zmk_hid_get_keyboard_report();
  size = sizeof(*zmk_hid_get_keyboard_report());
} else {
  data = get_boot_report();
  size = sizeof(*zmk_hid_get_boot_report());
}
```